src/Cache/Http/InvalidCacheTagsInterface.php

```diff
@@ -34,9 +34,9 @@ interface InvalidCacheTagsInterface
      *
      * @param EnvironmentInterface $environment The dc general environment.
      *
-     * @return InvalidCacheTags
+     * @return InvalidateCacheTags
      */
-    public function setEnvironment(EnvironmentInterface $environment): InvalidCacheTags;
```

Comment:
It has to return an instance of `InvalidCacheTagsInterface`

Reply:
The `InvalidCacheTagsInterface` is designed to be a service. Making the instance stateful by binding it to an instance of the environment doesn't look correct to me. I vote to drop this method at all and change the signature of `purgeCacheTags` to `public function purgeCacheTags(ModelInterface $model, EnvironmentInterface $environment): void`.

Reply:
Naming should rather be `InvalidateCacheTagsInterface` instead of `InvalidCacheTagsInterface`as the cache tags get used to invalidate instead of being invalid.

Reply:
fixed

Reply:
fix naming

Reply:
@dmolineus fix EnvironmentInterface